src/gsy_e/gsy_e_core/simulation/simulation.py

```diff
@@ -430,16 +427,26 @@ def _deactivate_areas(self, area: "AreaBase"):
         for child in area.children:
             self._deactivate_areas(child)
 
+    def _handle_external_communication(self):
```

Comment:
I like a lot that the external connection stuff are at least extracted to their own method! However, is it possible to not even call these methods using an early-return? For instance:

```suggestion
    def _handle_external_communication(self):
        if not self.config.external_connection_enabled:
            return
```
Would that work out ?

Reply:
That should work. All the methods inside check again if the aggregator is enabled, but why not returning early. Will do. Thanks!